server/src/main/java/org/opensearch/common/cache/Cache.java

```diff
@@ -424,68 +424,74 @@ public V computeIfAbsent(K key, CacheLoader<K, V> loader) throws ExecutionExcept
             }
         });
         if (value == null) {
-            // we need to synchronize loading of a value for a given key; however, holding the segment lock while
-            // invoking load can lead to deadlock against another thread due to dependent key loading; therefore, we
-            // need a mechanism to ensure that load is invoked at most once, but we are not invoking load while holding
-            // the segment lock; to do this, we atomically put a future in the map that can load the value, and then
-            // get the value from this future on the thread that won the race to place the future into the segment map
-            CacheSegment<K, V> segment = getCacheSegment(key);
-            CompletableFuture<Entry<K, V>> future;
-            CompletableFuture<Entry<K, V>> completableFuture = new CompletableFuture<>();
+            value = compute(key, loader);
+        }
+        return value;
+    }
 
-            try (ReleasableLock ignored = segment.writeLock.acquire()) {
-                future = segment.map.putIfAbsent(key, completableFuture);
-            }
+    public V compute(K key, CacheLoader<K, V> loader) throws ExecutionException {
```

Comment:
I need this to be public. 
OpenSearchOnHeapCache uses it directly. And this in turn used inside TieredSpilloverCache.computeIfAbsent.

Let me know if you still think if it's not needed.

Reply:
I can't use computeIfAbsent. Check TieredSpilloverCache.computeIfAbsent()
Here we first try to do a get on both tiers. If not present, we only do a compute() to avoid doing another get() internally.

Reply:
I have changed it to private. With recent changes, it was not needed anymore to be accessed from outside.